atecc608a/main.c

```diff
@@ -477,10 +440,12 @@ bool interactive_loop()
         return true;
     } else if (strcmp(command, "test") == 0) {
         run_tests();
-    } else if (strncmp(command, "generate_private", strlen("generate_private") - 1) == 0) {
+    } else if (strncmp(command, "generate_private",
```

Comment:
Why does this match not only `"generate_private"` but also e.g. `"generate_private_wibble"`?

Reply:
To match `"generate_private=5"` too, if an argument is to be supplied.

Reply:
Ok. It would be cleaner to only allow `"generate_private\0"` and `"generate_private="`, but that's not a problem. This is not the part of the example where we demonstrate how to use the API, so it's ok to take shortcuts here.